app/models/accounting/transaction.rb

```diff
@@ -95,4 +112,27 @@ def lookup_currency_id
       project.currency_id
     end
   end
+
+  def calculate_loan_amounts
+    loan_id = self.project_id
+
+    if self.loan_transaction_type == 'disbursement'
+      self.update_column(:change_in_principal, amount)
+    end
+
+    if amount <= interest_balance
+      self.change_in_interest =- amount
```

Comment:
This should be `change_in_interest = -amount`, the `self` is redundant, and it's important that it be clear that we're assigning "negative amount" -- right now it's easy for somebody reading over the code to confuse `=-` with the `-=` operator which would do `change_in_interest = change_in_interest - amount` which is not what we want.